app/src/main/java/com/example/vybe/MyRequestsActivity.java

```diff
@@ -1,60 +1,61 @@
 package com.example.vybe;
 
-import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.recyclerview.widget.DividerItemDecoration;
 import androidx.recyclerview.widget.LinearLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
 import android.os.Bundle;
-import android.util.Log;
-import android.widget.Toast;
 
 import com.example.vybe.Models.User;
-import com.google.android.gms.tasks.OnSuccessListener;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.firestore.CollectionReference;
 import com.google.firebase.firestore.DocumentSnapshot;
-import com.google.firebase.firestore.FieldValue;
 import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.FirebaseFirestoreException;
-import com.google.firebase.firestore.Query;
-import com.google.firebase.firestore.QueryDocumentSnapshot;
-import com.google.firebase.firestore.QuerySnapshot;
 
 import java.util.ArrayList;
 
 public class MyRequestsActivity extends AppCompatActivity {
 
     private static final String TAG = "MyRequestsActivity";
 
-    private ArrayList<User> requestList;
     private ProfileAdapter profileAdapter;
 
     private RecyclerView userRecyclerView;
     private FirebaseFirestore db = FirebaseFirestore.getInstance();
     private FirebaseAuth mAuth = FirebaseAuth.getInstance();
-    private String myUID = mAuth.getCurrentUser().getUid();
+
+    private RequestController requestController = RequestController.getInstance(MyRequestsActivity.this);
+    private ArrayList<String> myRequestList = requestController.getMyRequestList();
+    private ArrayList<User> userRequestList = requestController.getMyUserRequestList();
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_my_requests);
 
         userRecyclerView = findViewById(R.id.my_request_list);
+        profileAdapter = new ProfileAdapter(R.layout.user_item, userRequestList);
 
-        requestList = new ArrayList<>();
-
-        profileAdapter = new ProfileAdapter(R.layout.user_item, requestList);
         profileAdapter.setRequestClickListener(new ProfileAdapter.OnRequestClickListener() {
+
+            // TODO: move deleteItem calls from profileAdapter into controller? - use listener
             @Override
             public void onAcceptClick(int position) {
-                acceptFollowRequest(position);
+                User selectedUser = userRequestList.get(position);
+                String selectedID = selectedUser.getUserID();
+                requestController.acceptFollowRequest(selectedID);
+                profileAdapter.deleteItem(position);
+
```

Comment:
No. Move this back into a function called `acceptFollowRequest` but like just have it defined in this activity. Check how it looks like in master